apps/app/src/views/routes/Settings/components/users/EditRolesAction.tsx

```diff
@@ -0,0 +1,444 @@
+  mutation UpdateProviderProfile(
+    $providerId: ID!
+    $updateProviderProfileInput: UpdateProviderProfileInput!
+  ) {
+    updateProviderProfile(providerId: $providerId, input: $updateProviderProfileInput)
+  }
+`);
+
+const hasPrescriberRole = (roles: { value: string; label: string }[]) =>
+  roles.some((r) => r.label === 'Prescriber');
+
+const roleSchema = yup
+  .object({
+    email: yup
+      .string()
+      .email('Invalid email')
+      .required("Please include email of the person you'd like to invite"),
+    roles: rolesSchema.required().min(1, 'Must have at least one role'),
+    provider: yup
+      .object({
+        npi: yup
+          .string()
+          .required('NPI is required for prescribers')
+          .matches(/^[0-9]*$/, { message: 'Invalid NPI' }),
+        address: yup
+          .object({
+            street1: yup.string().required('Address is required'),
+            street2: yup.string(),
+            city: yup.string().required('City is required'),
+            state: yupStateSchema,
+            postalCode: yup
+              .string()
+              .required('Zip is required')
+              .matches(/^[0-9]{5}(?:-[0-9]{4})?$/, { message: 'Enter a valid zipcode' })
+          })
+          .required('Please enter an address'),
+        phone: yup
+          .string()
+          .matches(
+            /^\s*(?:\+?(\d{1,3}))?[-. (]*(\d{3})[-. )]*(\d{3})[-. ]*(\d{4})(?: *x(\d+))?\s*$/,
+            { message: 'Please enter a valid phone number' }
+          )
+          .required('Please enter a valid phone number')
+      })
+      .notRequired()
+      .default(undefined)
+      .when('roles', (roles: { value: string; label: string }[], schema) => {
+        return hasPrescriberRole(roles)
+          ? schema.required(
+              "Address, NPI and phone are required for users with 'Prescriber' permissions"
+            )
+          : schema.notRequired();
+      })
+  })
+  // If not a prescriber, set the provider to undefined
+  .transform((value) => {
+    if (hasPrescriberRole(value.roles)) {
+      return value;
+    }
+    return { ...value, provider: undefined };
+  });
+
+type RoleYupType = yup.InferType<typeof roleSchema>;
+type ProviderYupType = RoleYupType['provider'];
+type ProviderFormikTouchedType = FormikTouched<ProviderYupType>;
+type ProviderFormikErrorsType = FormikErrors<ProviderYupType>;
+
+export const EditRolesAction: React.FC<EditRolesActionProps> = ({ userId, onClose }) => {
+  const client = useClinicalApiClient();
+  const {
+    data: userData,
+    error: userDataError,
+    loading
+  } = useQuery(GetUserQuery, {
+    client,
+    variables: { userId: userId }
+  });
+  const [setUserRoles, { error }] = useMutation(SetUserRolesMutation, {
+    client,
+    refetchQueries: ['UsersListQuery', 'usersQuery']
+  });
+  const [updateProviderProfile] = useMutation(UpdateProviderProfileMutation, {
+    client,
+    refetchQueries: []
```

Comment:
This presumably should also refetch?

Reply:
At the very least, this should refetch `EditRolesActionGetUserQuery`

Reply:
UpdateProfile never happens by itself, UpdateRoles Refetches both.